src/background/environment.js

```diff
@@ -0,0 +1,25 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+export async function initializeEnvironment() {
+  // Issue #57: Disable login saving for extension pages.
+  const ourOrigin = browser.extension.getURL("").slice(0, -1);
+
+  // If login saving is enabled for us on startup, add an exception.
+  const savingEnabled = await browser.experiments.logins
+    .getLoginSavingEnabled(ourOrigin);
+  if (savingEnabled) {
+    await browser.experiments.logins
+      .setLoginSavingEnabled(ourOrigin, false);
+  }
+
+  // HACK: If our exception to login saving is removed, immediately re-add it.
+  browser.experiments.logins.onLoginSavingChanged
```

Comment:
Okay, here's the bit that could use some sanity checking. I worked out how to subscribe to the event that fires when a login saving exception is removed, and automatically re-add it. This may be a mistake, but I can't think of a reason why